File: 자료구조/Stack/MyStack.java
import java.util.ArrayList;

public class MyStack <T>{
    private ArrayList<T> stack = new ArrayList<>();

    public void push(T data){
        stack.add(data);
    }

    public T pop(){
        if (stack.isEmpty()) {
            return null;
        }
        return stack.remove(stack.size()-1);
    }

    public static void main(String[] args) {
        MyStack<Integer> myStack = new MyStack<>();
        myStack.push(1);
        myStack.push(2);
        System.out.println(myStack.pop());//2
        myStack.push(3);
        System.out.println(myStack.pop());//3
        System.out.println(myStack.pop());//1
    }
}

File: 자료구조/Stack/MyStack.kt

fun main(){
    val myStack = MyStack<Int>();
    myStack.push(1)
    myStack.push(2)
    println(myStack.pop())  //2
    myStack.push(3)
    println(myStack.pop())  //3
    println(myStack.pop())  //1
    println(myStack.pop())  //null
}

class MyStack <T>{
    val stack = mutableListOf<T>();
    fun push(data :T){
        stack.add(data)
    }

    fun pop():T?{
       if (stack.isEmpty()){
           return null
       }
        return stack.removeAt(stack.size-1)
    }

}